rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -924,6 +930,96 @@ void saveFederationElection_shouldSaveInStorage() {
+
+        StorageAccessor storageAccessor;
+        FederationStorageProvider federationStorageProvider;
+        ActivationConfig.ForBlock activations;
+
+        @BeforeEach
+        void setup() {
+            storageAccessor = new InMemoryStorage();
+            federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+            activations = ActivationConfigsForTest.all().forBlock(0L);
+        }
+
+        @Test
+        void saveFundTransactionUnsignedHash_preLovell700_shouldNotSaveInStorage() {
+
+            // Arrange
+
+            ActivationConfig.ForBlock arrowheadActivations = ActivationConfigsForTest.arrowhead600().forBlock(0L);
+
+            Sha256Hash expectedHash = BitcoinTestUtils.createHash(123_456_789);
+
+            federationStorageProvider.setFundTransactionUnsignedHash(expectedHash);
+
+            // Act
+
+            federationStorageProvider.save(networkParameters, arrowheadActivations);
+
+            // Assert
+
+            Sha256Hash actualHash = storageAccessor.getFromRepository(SVP_FUND_TX_HASH_UNSIGNED.getKey(), BridgeSerializationUtils::deserializeSha256Hash);
+
+            assertNull(actualHash);
+
+        }
+
+        @Test
+        void saveFundTransactionUnsignedHash_postLovell700_shouldSaveInStorage() {
+
+            // Arrange
+
+            Sha256Hash expectedHash = BitcoinTestUtils.createHash(123_456_789);
+
+            federationStorageProvider.setFundTransactionUnsignedHash(expectedHash);
+
+            // Act
+
+            federationStorageProvider.save(networkParameters, activations);
+
+            // Assert
+
+            Sha256Hash actualHash = storageAccessor.getFromRepository(SVP_FUND_TX_HASH_UNSIGNED.getKey(), BridgeSerializationUtils::deserializeSha256Hash);
+
+            assertEquals(expectedHash, actualHash);
+
+        }
+
+        @Test
+        void saveFundTransactionUnsignedHash_postLovell700AndResettingToNull_shouldSaveNullInStorage() {
+
+            // Arrange
+
+            Sha256Hash expectedHash = BitcoinTestUtils.createHash(123_456_789);
+
+            // Initially setting a valid hash in storage
+            federationStorageProvider.setFundTransactionUnsignedHash(expectedHash);
+            federationStorageProvider.save(networkParameters, activations);
+            Sha256Hash originalHashInStorage = storageAccessor.getFromRepository(
+                SVP_FUND_TX_HASH_UNSIGNED.getKey(), BridgeSerializationUtils::deserializeSha256Hash);
+
+            // Act
+
+            // Setting to null
+            federationStorageProvider.setFundTransactionUnsignedHash(null);
+            federationStorageProvider.save(networkParameters, activations);
+
+            // Assert
+
+            Sha256Hash finalHashFromStorage = storageAccessor.getFromRepository(SVP_FUND_TX_HASH_UNSIGNED.getKey(), BridgeSerializationUtils::deserializeSha256Hash);
+
+            assertEquals(expectedHash, originalHashInStorage);
```

Comment:
is this assertion needed? is covered in the previous test right?

Reply:
Updated.